cmd/demoserver/main_test.go

```diff
@@ -70,13 +69,15 @@ func TestElizaServer(t *testing.T) {
 			stream := client.Converse(context.Background())
 			var wg sync.WaitGroup
 			wg.Add(2)
+			errs := make(chan error, 4)
```

Comment:
This would be much simpler to use an `errgroup.Group` instead of `sync.WaitGroup` and all of this error-tracking channel chicanery.

```go
grp, ctx := errgroup.WithContext(context.Background())
// use ctx to start RPC ...
grp.Go(func() error {
    // send messages, immediately return err on failure
    // (instead of storing in channel)
})
grp.Go(func() error {
    // receive messages, immediately return err on failure
})
require.NoError(t, grp.Wait())
```

Reply:
Big fan of the line `error-tracking channel chicanery.`. Updated lmk what you think.